src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -324,8 +324,8 @@ def set_active_subscription(self, subscription):  # take id or name
                   x[_ENVIRONMENT_NAME] == active_cloud.name]
 
         if len(result) != 1:
-            raise CLIError("The subscription of '{}' does not exist or has more than"
-                           " one match in cloud '{}'.".format(subscription, active_cloud.name))
+            raise CLIError("The subscription of '{}' {} in cloud '{}'.".format(
+                subscription, "doesn't exist" if not result else 'has more than one match', active_cloud.name))
```

Comment:
nit for clarity?: `'has more than one match' if result else "doesn't exist"`

Reply:
i will address it later